storage/local/series.go

```diff
@@ -459,13 +459,21 @@ func (s *memorySeries) preloadChunksForRange(
 	}
 	if s.chunkDescsOffset != 0 && from.Before(firstChunkDescTime) {
 		cds, err := mss.loadChunkDescs(fp, s.persistWatermark)
+		if s.chunkDescsOffset != -1 && len(cds) != s.chunkDescsOffset {
```

Comment:
If there is an error already returned by `mss.loadChunkDescs` it seems more that that should be the only relevant error and this one here would just be a red herring.